vdr/src/types/transaction.rs

```diff
@@ -495,3 +495,256 @@ pub struct BlockDetails {
+
+#[cfg(test)]
+pub mod test {
+    use super::*;
+    use crate::{
+        client::client::test::{mock_client, write_transaction, INVALID_ADDRESS},
+        contracts::network::test::{
+            ADD_VALIDATOR_METHOD, VALIDATOR_ADDRESS, VALIDATOR_CONTROL_NAME, VALIDATOR_LIST_BYTES,
+        },
+    };
+
+    const INVALID_METHOD: Option<&str> = Some(INVALID_ADDRESS);
+    const INVALID_CONTRACT: Option<&str> = Some(INVALID_ADDRESS);
+    const INVALID_ACC_ADDRESS: Option<&str> = Some(INVALID_ADDRESS);
+    const CONTRACT_METHOD_EXAMPLE: Option<&str> = Some(ADD_VALIDATOR_METHOD);
+    const CONTRACT_NAME_EXAMPLE: Option<&str> = Some(VALIDATOR_CONTROL_NAME);
+
+    fn matches_error_type(actual_error: VdrError, expected_error: VdrError) -> bool {
+        std::mem::discriminant(&actual_error) == std::mem::discriminant(&expected_error)
+    }
+
+    #[cfg(test)]
+    pub mod txn_test {
+        use super::*;
+
+        #[async_std::test]
+        async fn get_to_invalid() {
+            let transaction = Transaction {
+                to: Address::from(INVALID_ADDRESS),
+                ..write_transaction()
+            };
+
+            let get_to_err = transaction.get_to().unwrap_err();
+
+            assert!(matches!(
+                get_to_err,  | VdrError::ClientInvalidTransaction { .. }
+            ));
+        }
+
+        #[async_std::test]
+        async fn get_nonce_not_set() {
+            let transaction = Transaction {
+                nonce: None,
+                ..write_transaction()
+            };
+
+            let get_nonce_err = transaction.get_nonce().unwrap_err();
+
+            assert!(matches!(
+                get_nonce_err,  | VdrError::ClientInvalidTransaction { .. }
+            ));
+        }
+
+        #[async_std::test]
+        async fn get_transaction_signature_not_set() {
+            let transaction = Transaction {
+                signature: RwLock::new(None),
+                ..write_transaction()
+            };
+
+            let get_sig_err = transaction.get_transaction_signature().unwrap_err();
+
+            assert!(matches!(
+                get_sig_err,  | VdrError::ClientInvalidTransaction { .. }
+            ));
+        }
+
+        #[async_std::test]
+        async fn get_transaction_signature_invalid() {
+            let transaction = Transaction {
+                signature: RwLock::new(Some(TransactionSignature {
+                    v: 1,
+                    r: vec![1; 32],
+                    s: vec![1; 32],
+                })),
+                ..write_transaction()
+            };
+
+            let get_sig_err = transaction.get_transaction_signature().unwrap_err();
+
+            assert!(matches!(
+                get_sig_err,  | VdrError::ClientInvalidTransaction { .. }
+            ));
+        }
+    }
+
+    #[cfg(test)]
+    pub mod txn_builder_test {
+        use super::*;
+        use rstest::rstest;
+        use std::ops::Deref;
+
+        async fn setup_transaction_builder(
```

Comment:
we do not need this function anymore.  Just move it to test body

Reply:
Moved